superset-frontend/src/dashboard/components/FiltersBadge/index.tsx

```diff
@@ -0,0 +1,127 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import cx from 'classnames';
+import { connect, Dispatch } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import Icon from 'src/components/Icon';
+import { setDirectPathToChild } from 'src/dashboard/actions/dashboardState';
+import DetailsPanelPopover, { Indicator } from './DetailsPanel';
+import { Pill } from './Styles';
+import { selectIndicatorsForChart, IndicatorStatus } from './selectors';
+
+const mapDispatchToProps = (dispatch: Dispatch<any>) => {
```

Comment:
In `Dashboard` directory, we prefer to keep redux container component in `containers` subdirectory, and `components` subdirectory only have unconnected React components.